NohanaImagePicker/AssetDateSectionCreater.swift

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2021 nohana, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Photos
+
+final class AssetDateSectionCreater {
+    func createSections(assetList: PHAssetCollection, options: PHFetchOptions) -> [AssetDateSection] {
+        var albumDateSectionList = [AssetDateSection]()
+        let fetchAssetlist = PHAsset.fetchAssets(in: assetList, options: options)
+        let allAssets = fetchAssetlist.objects(at: IndexSet(0..<fetchAssetlist.count))
+        let calender = Calendar.current
+        var assetsByDate = [(DateComponents, [PHAsset])]()
+        var assetsByDateIndex = 0
+        for asset in allAssets {
+            if  assetsByDateIndex > 0 {
+                if assetsByDate[assetsByDateIndex - 1].0 == calender.dateComponents([.day, .year, .month], from: (asset.creationDate)!) {
+                    assetsByDate[assetsByDateIndex - 1].1.append(asset)
+                } else {
+                    let value = (calender.dateComponents([.day, .year, .month], from: (asset.creationDate)!), [asset])
+                    assetsByDate.append(value)
+                    assetsByDateIndex += 1
+                }
+            } else if assetsByDate.count == assetsByDateIndex {
+                let value = (calender.dateComponents([.day, .year, .month], from: (asset.creationDate)!), [asset])
+                assetsByDate.append(value)
+                assetsByDateIndex += 1
+            }
+        }
+        albumDateSectionList = assetsByDate.map { AssetDateSection(creationDate: calender.date(from: $0.0) ?? Date(timeIntervalSince1970: 0), assetResult: $0.1) }
```

Comment:
It may be better to use "init (grouping: by :)" of "Dictionary".
https://developer.apple.com/documentation/swift/dictionary/3127163-init

Code example
```swift
let dateFormatter = DateFormatter()
dateFormatter.dateStyle = .long
dateFormatter.timeStyle = .none

albumDateSectionList = Dictionary(grouping: allAssets, by: { asset in
    let dateString = dateFormatter.string(from: asset.creationDate!)
    return dateFormatter.date(from: dateString)!
}).reduce(into: [AssetDateSection](), {
    $0.append(AssetDateSection(creationDate: $1.key, assetResult: $1.value))
}).sorted(by: {
    $0.creationDate < $1.creationDate
})
```

Reply:
I was also implemented `Dictionary`. 
but `Array` are faster than `Dictionary`.

I think that it should be a fast implementation because it is stressful for the user to take a long time to load.